Public/Src/Utilities/UnitTests/Executables/TestProcess/Operation.cs

```diff
@@ -787,6 +797,11 @@ private void DoWriteFile(string file, string content)
         {
             try
             {
+                if (AdditionalArgs == AllUppercasePath)
+                {
+                    file = file.ToUpperInvariant();
+                }

```

Comment:
For Windows, it's a bit weird that we have a test that exercises casing because if the file entry (could be a directory name) is already there, then even though we write with all upper case, the directory name is still the one when it was first created.

Reply:
This is making sure that tracking done on detours side for paths were write allowed are checked is actually case insensitive. E.g. the pip first creates 'FOO.txt', and then writes again in 'foo.txt'. If the tracking was case sensitive we will think that the second write is a first write on an existing file, and therefore deny it.

---
In reply to: [299192998](https://github.com/microsoft/BuildXL/pull/536#discussion_r299192998) [](ancestors = 299192998)

Reply:
Can this kind of test be done in SandboxedProcessPipExecutorTest? 

---
In reply to: [299194088](https://github.com/microsoft/BuildXL/pull/536#discussion_r299194088) [](ancestors = 299194088,299192998)

Reply:
I think that's possible, but a little bit harder to implement. I'm currently using Operation, which is pretty handy, and the corresponding test (WritingToExistentFileProducedBySamePipIsAllowed) complements some aspects of other side-by-side tests, that require the integration test machinery. So unless you are strong about it, I'd prefer to leave it as is.

---
In reply to: [299202987](https://github.com/microsoft/BuildXL/pull/536#discussion_r299202987) [](ancestors = 299202987,299194088,299192998)